03-transactions.md

```diff
@@ -1423,6 +1432,401 @@ And, here are the test vectors themselves:
+    num_htlcs: 5
+    # signature for output 0 (HTLC 0)
+    remote_htlc_signature = 304402206a6e59f18764a5bf8d4fa45eebc591566689441229c918b480fb2af8cc6a4aeb02205248f273be447684b33e3c8d1d85a8e0ca9fa0bae9ae33f0527ada9c162919a6
+    # signature for output 1 (HTLC 2)
+    remote_htlc_signature = 3045022100d5275b3619953cb0c3b5aa577f04bc512380e60fa551762ce3d7a1bb7401cff9022037237ab0dac3fe100cde094e82e2bed9ba0ed1bb40154b48e56aa70f259e608b
+    # signature for output 2 (HTLC 1)
+    remote_htlc_signature = 304402201b63ec807771baf4fdff523c644080de17f1da478989308ad13a58b51db91d360220568939d38c9ce295adba15665fa68f51d967e8ed14a007b751540a80b325f202
+    # signature for output 3 (HTLC 3)
+    remote_htlc_signature = 3045022100daee1808f9861b6c3ecd14f7b707eca02dd6bdfc714ba2f33bc8cdba507bb182022026654bf8863af77d74f51f4e0b62d461a019561bb12acb120d3f7195d148a554
+    # signature for output 4 (HTLC 4)
+    remote_htlc_signature = 304402207e0410e45454b0978a623f36a10626ef17b27d9ad44e2760f98cfa3efb37924f0220220bd8acd43ecaa916a80bd4f919c495a2c58982ce7c8625153f8596692a801d
+    # local_signature = 304402207cb324fa0de88f452ffa9389678127ebcf4cabe1dd848b8e076c1a1962bf34720220116ed922b12311bd602d67e60d2529917f21c5b82f25ff6506c0f87886b4dfd5
+    output htlc_success_tx 0: 020000000001018154ecccf11a5fb56c39654c4deb4d2296f83c69268280b94d021370c94e219700000000000000000001e8030000000000002200204adb4e2f00643db396dd120d4e7dc17625f5f2c11a40d857accc862d6b7dd80e050047304402206a6e59f18764a5bf8d4fa45eebc591566689441229c918b480fb2af8cc6a4aeb02205248f273be447684b33e3c8d1d85a8e0ca9fa0bae9ae33f0527ada9c162919a60147304402207cb324fa0de88f452ffa9389678127ebcf4cabe1dd848b8e076c1a1962bf34720220116ed922b12311bd602d67e60d2529917f21c5b82f25ff6506c0f87886b4dfd5012000000000000000000000000000000000000000000000000000000000000000008a76a91414011f7254d96b819c76986c277d115efce6f7b58763ac67210394854aa6eab5b2a8122cc726e9dded053a2184d88256816826d6231c068d4a5b7c8201208763a914b8bcb07f6344b42ab04250c86a6e8b75d3fdbbc688527c21030d417a46946384f88d5f3337267c5e579765875dc4daca813e21734b140639e752ae677502f401b175ac686800000000
```

Comment:
This is wrong?  It should be spending the commit_tx above, but it isn't.  bitcoin-cli decodes this:

```
{
  "txid": "b8afcdfb3e0b49be541a0c24569f49d9ebd03d275f809780d0afde66bec01da2",
  "hash": "402ac1f6656f1f9adbff78d7191a4f30192390bdc2db187e3de1ca9de1c79c68",
  "version": 2,
  "size": 414,
  "vsize": 174,
  "weight": 696,
  "locktime": 0,
  "vin": [
    {
      "txid": "97214ec97013024db9808226693cf896224deb4d4c65396cb55f1af1ccec5481",
      "vout": 0,
```

Reply:
This is embarrassing, I have no idea how I managed to get that wrong...
These test vectors didn't even match what's in eclair, so I messed up either a copy-paste or a git rebase :thinking: 
That should be fixed in https://github.com/lightningnetwork/lightning-rfc/pull/539/commits/c6d13fb3f9196464d842dc579723cf6de35ec8f8

Reply:
@rustyrussell would you have some time to validate these? I think they should work this time :sweat_smile: